accepted/2020/mono-convergence/security-support-new-platforms.md

```diff
@@ -0,0 +1,90 @@
+# Security support for new .NET 5 platforms
+
+## Overview
+Currently in .NET Core, we have a solid [strategy](https://github.com/dotnet/corefx/blob/master/Documentation/architecture/cross-platform-cryptography.md) for dealing with cryptography on Linux and OSX.  At the native level, `System.Security.Cryptography.Native` relies upon OpenSSL for Linux and `System.Security.Cryptography.Native.Apple` relies upon platform specific API’s for OSX.  
+
+With the introduction of Android, iOS, and Wasm workloads, there are implications to consider that may alter our strategy and require additional work.  This document seeks to describe the implications and what options we may have.
+
+## Security Dependencies
+For `System.Security.Cryptography.Native` (Linux):
+
+[src/libraries/Native/Unix/System.Security.Cryptography.Native] (https://github.com/dotnet/runtime/tree/master/src/libraries/Native/Unix/System.Security.Cryptography.Native)
+
+For `System.Security.Cryptography.Native.Apple` (OSX):
+
+[src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple] (https://github.com/dotnet/runtime/tree/master/src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple)
+
+### Android
+There is currently no platform level API’s that can work with Android, so any native dependency we take must be bundled with the individual application.
+
+**System.Security.Cryptography.Native**
+
+The current dependency on OpenSSL could work as there seems to be support for building on Android.  This is nice because it would seemingly ‘just fit’ with what we already have.  However, there are some things to take into consideration:
+
+* Do we fork a copy of OpenSSL and build from that?
+* What is our process for shipping security vulnerability updates?
```

Comment:
If we have to ship something with an app, don't these questions/concerns apply to any choice?

Reply:
Yep - I'll try to restate.

Reply:
I was aware that Google Play Services includes cronet & boringSSL, but they seem to only be exposed via a Java cronet API - https://developer.android.com/guide/topics/connectivity/cronet - which probably doesn't help us.

Reply:
Last I knew, Google's documented stance on BoringSSL was that it was unsupported, could change its ABI at any time, etc... exactly what you want when taking a dependency on a security library ;-)  If there's a supported/maintained variant, that's then more interesting.

Reply:
We really, really, really want to avoid shipping a cryptography library with the product.  AFAIK, it adds a whole lot of government paperwork; and it creates a very complicated servicing model for both us and the applications built using it.

If there's any library in the box that we can reliably use, that's the best choice almost without regard to how painful it is to initially use it.

Reply:
I think you'd need to use the conscrypt APIs - which are higher level http APIs.
This may help: https://groups.google.com/forum/#!topic/conscrypt/T4c2YCY91qc.